support-workers/src/main/scala/com/gu/support/workers/lambdas/UpdateSupporterProductData.scala

```diff
@@ -38,14 +39,19 @@ class UpdateSupporterProductData(serviceProvider: ServiceProvider)
     supporterDataDynamoService: SupporterDataDynamoService
   ) = {
     getSupporterRatePlanItemFromState(state, catalogService) match {
-      case Some(supporterRatePlanItem) => supporterDataDynamoService.writeItem(supporterRatePlanItem)
-      case None => Future.failed(
+      case Right(Some(supporterRatePlanItem)) => supporterDataDynamoService.writeItem(supporterRatePlanItem).map(_ => ())
+      case Right(None) => Future.successful(())
+      case Left(_) => Future.failed(
```

Comment:
I wouldn't go back and add it now, but might have been a nice place to alias Right/None with some vals